pkg/workloadapi/workload.proto

```diff
@@ -222,6 +222,7 @@ message GatewayAddress {
   }
   // port to reach the gateway at
   uint32 port = 3;
+  PortList ports = 4;
```

Comment:
I think we will want the ports named so we know what they're used for?

perhaps `port` remains the mtls port and we have a new `oneway_tls_port` or something to that effect as the new field?

Reply:
Building on Kevin's comment, iirc as long as the field number and type remain consistent you may rename a field in a protobuf safely so we could consider something like:

```proto
  // port to reach the gateway at for mtls HBONE connections
  uint32 hbone_mtls_port = 3;
  // port to reach the gateway at for single tls HBONE connections
  // used for traffic originating outside the mesh reaching a waypoint-enabled destination
  uint32 hbone_single_tls_port = 4;
```

Reply:
Love it; good call

Reply:
does single tls means not require client certificates? 

There is a `ServerTLSSettings_SIMPLE` used by istio gateway. maybe we should keep consistent.

By the way, recently there is a new mode  added `ServerTLSSettings_OPTIONAL_MUTUAL`, which can work in any case.
Can we use that mode?

Reply:
Yeah that's the overall question: multiple ports vs. 1 port with a boolean knob for what kind of TLS is allowed. Slack thread for context: https://istio.slack.com/archives/C049TCZMPCP/p1694037013990319

Reply:
@linsun's comment altered my thinking on this proto slightly. Should we add `optional` to `hbone_single_tls_port` since a strict waypoint likely won't have any "single tls" listener?

Reply:
Yeah that's fair